### PR TITLE
TransposeSinking PRelu special slope fix input(1) 

### PR DESCRIPTION
**Pull request by evkotov**
### Details:
 - fix TSBinaryBackward transformation
 - add unit tests

### Tickets:
 - 102236

### CODE REVIEW

File: src/common/transformations/src/transformations/transpose_sinking/ts_binary.cpp [hunk 1]
```diff
@@ -30,6 +30,48 @@ TSBinaryForward::TSBinaryForward() : TSForwardBase() {
...
+ * the second dimension of data input, then per channel broadcast is applied.
+ * In such a case we need to insert Unsqueeze before Transpose with another axes.
+ */
+bool IsSpecialPRelu(NodePtr node) {
+    auto prelu = as_type_ptr<ov::op::v0::PRelu>(node);
+    if (!prelu)
+        return false;
+
+    const auto slope_shape = prelu->get_input_partial_shape(1);
+    if (slope_shape.is_dynamic() || slope_shape.size() != 1)
```
Comment (itikhono) [inline src/common/transformations/src/transformations/transpose_sinking/ts_binary.cpp:63]: t's more precise to check the rank and the concrete dimension of "arg":
```
const auto arg_shape = prelu->get_input_partial_shape(0);
const auto arg_rank = arg_shape.get_rank();
if (rank.is_dynamic()) {
   return false;
}

const auto channel_dim_idx =arg_rank .get_lenght() > 1 ? 1 : 0;
if (arg_shape[channel_dim_idx].is_dynamic()) 
    return false;
return arg_shape[channel_dim_idx].get_lenght() == slope_shape.get_shape()[0];
```

File: src/common/transformations/src/transformations/transpose_sinking/ts_binary.cpp [hunk 2]
```diff
@@ -30,6 +30,48 @@ TSBinaryForward::TSBinaryForward() : TSForwardBase() {
...
+        return false;
+
+    const auto arg_shape = prelu->get_input_partial_shape(0);
+    if (arg_shape.is_dynamic())
+        return false;
+
+    const auto channel_dim_idx = arg_shape.size() > 1 ? 1 : 0;
```
Comment (itikhono) [inline src/common/transformations/src/transformations/transpose_sinking/ts_binary.cpp:70]: do we have a test when arg shape has rank = 1 and slope shape has rank = 1?

File: src/common/transformations/src/transformations/transpose_sinking/ts_binary.cpp [hunk 3]
```diff
@@ -30,6 +30,48 @@ TSBinaryForward::TSBinaryForward() : TSForwardBase() {
     transpose_sinking(matcher_name);
 }
 
+namespace {
+using NodePtr = std::shared_ptr<Node>;
+/**
+ * Inserts Unsqueeze node as a child to @arg node with axes {0, 2, ... N - 1}, where N = @arg n_dims
+ */
+NodePtr InsertBroadcastUnsqueezePReluSlope(const Output<Node>& node, size_t n_dims) {
```
Comment (itikhono) [inline src/common/transformations/src/transformations/transpose_sinking/ts_binary.cpp:38]: not so sure it works for arg rank = 1 and slope rank = 1 and arg_dim == slope_dim case

File: src/common/transformations/src/transformations/transpose_sinking/ts_binary.cpp (hunk 2 above)
Comment (evkotov) [reply src/common/transformations/src/transformations/transpose_sinking/ts_binary.cpp:70]: it works

File: src/common/transformations/src/transformations/transpose_sinking/ts_binary.cpp (hunk 3 above)
Comment (evkotov) [reply src/common/transformations/src/transformations/transpose_sinking/ts_binary.cpp:38]: I add unit test. It works

File: src/common/transformations/src/transformations/transpose_sinking/ts_binary.cpp [hunk 4]
```diff
@@ -30,6 +30,48 @@ TSBinaryForward::TSBinaryForward() : TSForwardBase() {
...
+    const auto slope_shape = prelu->get_input_partial_shape(1);
+    if (slope_shape.is_dynamic() || slope_shape.size() != 1)
+        return false;
+
+    const auto arg_shape = prelu->get_input_partial_shape(0);
+    if (arg_shape.rank().is_dynamic())
+        return false;
+
+    const auto channel_dim_idx = arg_shape.size() > 1 ? 1 : 0;
+    return arg_shape.get_shape()[channel_dim_idx] == slope_shape.get_shape()[0];
```
Comment (itikhono) [inline src/common/transformations/src/transformations/transpose_sinking/ts_binary.cpp:71]: you can't use arg_shape.get_shape() if we assume that arg shape can be dynamic
you have to use .get_partial_shape() and also add a check that arg_shape.get_partial_shape()[channel_dim_idx].is_dynamic

**Review by itikhono — APPROVED** [review]

**Pull request merged**